### PR TITLE
Feature/8 sensitive data filter

### PR DESCRIPTION
**Pull request by Toumash**
Description: #8

### CODE REVIEW

File: ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Options/BodyLoggerOptions.cs [hunk 1]
```diff
@@ -45,5 +45,23 @@ public BodyLoggerOptions()
         ///     Defines the text to append in case the body should be truncated <seealso cref="MaxBytes"/>
         /// </summary>
         public string Appendix { get; set; } = "\n---8<------------------------\nTRUNCATED DUE TO MAXBYTES LIMIT";
+
+        public IEnumerable<string> PropertyNamesWithSensitiveData { get; set; } = new List<string>()
```
Comment (matthiasguentert) [inline ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Options/BodyLoggerOptions.cs:49]:
```suggestion
        public List<string> PropertyNamesWithSensitiveData { get; set; } = new List<string>()
```

So the user can directly call `Add()` when configuring the middleware.

File: ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Options/BodyLoggerOptions.cs [hunk 2]
```diff
@@ -45,5 +45,23 @@ public BodyLoggerOptions()
...
+            "passwd",
+            "api_key",
+            "access_token",
+            "accessToken",
+            "auth",
+            "credentials",
+            "mysql_pwd"
+        };
+
+        public IEnumerable<string> SensitiveDataRegexes { get; set; } = new List<string>()
```
Comment (matthiasguentert) [inline ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Options/BodyLoggerOptions.cs:62]:
```suggestion
        public List<string> SensitiveDataRegexes { get; set; } = new List<string>()
```
So the user can directly call `Add()` when configuring the middleware.

File: ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Filters/ISensitiveDataFilter.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Azureblue.ApplicationInsights.RequestLogging.Filters
```
Comment (matthiasguentert) [inline ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Filters/ISensitiveDataFilter.cs:1]:
```suggestion
﻿namespace Azureblue.ApplicationInsights.RequestLogging
```
May I suggest to keep all in a single namespace to shorten the list of `using` statements?

File: ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Filters/SensitiveDataFilter.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+
+namespace Azureblue.ApplicationInsights.RequestLogging.Filters
```
Comment (matthiasguentert) [inline ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Filters/SensitiveDataFilter.cs:8]:
```suggestion
namespace Azureblue.ApplicationInsights.RequestLogging
```

File: ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Filters/SensitiveDataFilter.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            }
+        }
+
+        private void RemoveFromArray(JsonArray jArray)
+        {
+            if (jArray == null) throw new ArgumentNullException(nameof(jArray));
+
+            foreach (var jNode in jArray.Where(v => v != null))
+            {
+                RemoveIds(jNode);
```
Comment (matthiasguentert) [inline ApplicationInsightsRequestLogging/ApplicationInsightsRequestLogging/Filters/SensitiveDataFilter.cs:93]: Actually, I am not clear about the implications of a possibly null reference argument here as raised by the compiler.

File: ApplicationInsightsRequestLogging/ApplicationInsightsRequestLoggingTests/SensitiveDataFilter/SensitiveDataFilterTests.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
+using Azureblue.ApplicationInsights.RequestLogging.Filters;
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Xunit;
+
+namespace ApplicationInsightsRequestLoggingTests.Reader
```
Comment (matthiasguentert) [inline ApplicationInsightsRequestLogging/ApplicationInsightsRequestLoggingTests/SensitiveDataFilter/SensitiveDataFilterTests.cs:8]:
```suggestion
namespace ApplicationInsightsRequestLoggingTests.Filters
```

File: ApplicationInsightsRequestLogging/ApplicationInsightsRequestLoggingTests/BodyLoggerMiddlewareTests.cs [hunk 7]
```diff
@@ -11,6 +11,7 @@
 using Moq;
 using Microsoft.Extensions.Options;
 using Microsoft.Extensions.DependencyInjection;
+using Azureblue.ApplicationInsights.RequestLogging.Filters;
```
Comment (matthiasguentert) [inline ApplicationInsightsRequestLogging/ApplicationInsightsRequestLoggingTests/BodyLoggerMiddlewareTests.cs:14]:
```suggestion
```

**Review by matthiasguentert — COMMENTED** [review]
Thank you very much for your contribution. I allowed myself to annotate some minor findings. What do you think?

**Comment by lordscarlet** [comment]
Is there a plan to document how to use this and publish it to nuget?

**Comment by ColinNg** [comment]
> Is there a plan to document how to use this and publish it to nuget?

I haven't seen any activity in this repo, so in addition to the Pull Request #19, I've rebased the code and published it to NuGet as [Azureblue.ApplicationInsights.RequestLogging-with-Sensitive-Data-Filter](https://www.nuget.org/packages/Azureblue.ApplicationInsights.RequestLogging-with-Sensitive-Data-Filter/) which I would be happy to take down if PR 19 gets approved/merged.

**Comment by matthiasguentert** [comment]
> > Is there a plan to document how to use this and publish it to nuget?
> 
> I haven't seen any activity in this repo, so in addition to the Pull Request #19, I've rebased the code and published it to NuGet as [Azureblue.ApplicationInsights.RequestLogging-with-Sensitive-Data-Filter](https://www.nuget.org/packages/Azureblue.ApplicationInsights.RequestLogging-with-Sensitive-Data-Filter/) which I would be happy to take down if PR 19 gets approved/merged.

Dear @ColinNg, thanks a bunch for your contributions. I'll have a look this week and get back!

**Comment by Toumash** [comment]
Closing this one then

**Pull request closed without merging**